### DISCUSSION TITLE
I'm having trouble allowing imports from `astro:*` prefixed paths.

### DISCUSSION THREAD

**Discussion by nomadshiba** [Igorkowalski94/eslint-plugin-project-structure#23] [Help] [answered]
I was able to allow imports from `node:*` prefixed paths by doing this:

```js
pathAliases: {
  baseUrl: "./",
  paths: {
      "node:*": ["*"]
  }
}
```

But no matter what I do I can't seem to allow imports from `astro:*` prefixed paths.
I tried using `pathAliases`, i tried to adding it to `allowImportsFrom`, nothing works.

**Comment by Igorkowalski94** [comment]
Could you show me what the error displays when you have [debugMode ](https://github.com/Igorkowalski94/eslint-plugin-project-structure/wiki/project%E2%80%91structure-%E2%80%8Bindependent%E2%80%91modules#debug-mode) enabled?

**Comment by nomadshiba** [comment]
```
🔥 Cannot find module. If the import includes a path alias, make sure that you have added the alias to the configuration. 🔥eslintproject-structure/independent-modules
```

**Reply by Igorkowalski94** [reply]
What does an example import look like in Astro, and what exactly does its alias refer to in the project?

**Reply by Igorkowalski94** [reply]
```tsx
// astro = ./src
// src/components/ComponentName.tsx
import x from "astro/components/ComponentName"
```
``` json
 "pathAliases": {
    "baseUrl": ".",
    "paths": {
      "astro/*": ["src/*"]
    }
  }
```

**Reply by nomadshiba** [reply]
None of the `astro:*` prefixed imports work.
```ts
import { defineMiddleware } from "astro:middleware";
import { number, object, string } from "astro:schema";
import {} from "astro:actions";
import {} from "astro:i18n";
import {} from "astro:transitions";
// etc...
```

Since `astro:middleware` also available with `astro/middleware`, I tried to add path alias to it, testing to see at least if that gets fixed, which also didn't work:
```js
paths: {
   "astro:*": ["astro/*"],
}
```

Weird thing is this also gives the same error:
```ts
import { defineMiddleware } from "astro/middleware";
```

It's in `node_modules`

**Reply by nomadshiba** [reply]
Just realized, `import { getEnv } from "astro/env/runtime";` this also has the same issue.

**Reply by Igorkowalski94** [reply]
In that case, it's not an issue with the path aliases. It looks like the library doesn't recognize imports from Astro as external imports. Do you have a project where I could debug the issue?

**Reply by Igorkowalski94** [reply]
If you make such an import without specifying path aliases in the config, it still doesn't find the module, right? Even after deleting the cache or closing and reopening the IDE?

```ts
import { defineMiddleware } from "astro/middleware";

```

**Reply by nomadshiba** [reply]
Yes.
I'm creating a fresh project, to test it.

**Reply by nomadshiba** [reply]
https://github.com/DeepDoge/issue-astro

**Reply by Igorkowalski94** [reply]
Thanks, I'll look into it and let you know once I find the cause or release a version with the fix.

**Reply by Igorkowalski94** [reply, marked as answer]
https://github.com/Igorkowalski94/eslint-plugin-project-structure/releases/tag/v3.10.4  @DeepDoge

**Reply by Igorkowalski94** [reply]
And you can remove `pathAliases` from your independentModules config.

**Reply by Igorkowalski94** [reply]
@DeepDoge If everything works correctly, please mark this [comment ](https://github.com/Igorkowalski94/eslint-plugin-project-structure/discussions/23#discussioncomment-11351134) as the answer.
